### ISSUE TITLE
Attempting to build custom package throws error

### ISSUE THREAD

**Issue by tjefferson401** [pyodide/pyodide#2599]
## 🐛 Bug

Running `python -m pyodide_build buildpkg meta.yaml` in order to build a custom packages, yields the error "KeyError: 'HOSTINSTALLDIR'"

This happens with every package, including built ins

Another discussion raised a similar issue, where the recommendation was to use "make" instead, however that yields the same error

Macbook pro monterrey 

Pyodide latest

**Comment by hoodmane** [comment]
Could you give the full traceback? Also are you working in the docker image or not?

**Comment by tjefferson401** [comment]
I was not working on docker image. will attempt that. And traceback is below, thanks!
Traceback:
```
Traceback (most recent call last):
  File "/Users/thomasj/opt/anaconda3/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/thomasj/opt/anaconda3/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/thomasj/opt/anaconda3/lib/python3.8/site-packages/pyodide_build/__main__.py", line 62, in <module>
    main()
  File "/Users/thomasj/opt/anaconda3/lib/python3.8/site-packages/pyodide_build/__main__.py", line 44, in main
    HOSTINSTALLDIR = os.environ["HOSTINSTALLDIR"]
  File "/Users/thomasj/opt/anaconda3/lib/python3.8/os.py", line 675, in __getitem__
    raise KeyError(key) from None
KeyError: 'HOSTINSTALLDIR'
```

**Comment by hoodmane** [comment]
The error is presumably coming from inside `get_make_flag`. Tracing out calls to `get_make_flag` shows that the following calls are made:
```
HOSTSITEPACKAGES
SIDE_MODULE_CFLAGS
SIDE_MODULE_CXXFLAGS
SIDE_MODULE_LDFLAGS
TARGETINSTALLDIR
HOSTINSTALLDIR
```
odd that all but the last one works. In particular, `HOSTSITEPACKAGES` and `HOSTINSTALLDIR` are defined on consecutive lines of `Makefile.envs:
```
export HOSTINSTALLDIR=$(PYODIDE_ROOT)/packages/.artifacts
export HOSTSITEPACKAGES=$(PYODIDE_ROOT)/packages/.artifacts/lib/python$(PYMAJOR).$(PYMINOR)/site-packages
```
not clear why it would successfully access the first but then not successfully access the second.

**Comment by hoodmane** [comment]
@tjefferson401 For one thing, you must use Python 3.10. I guess we should put in a version check.

**Comment by hoodmane** [comment]
Also, which branch / commit are you on? I can't find one with `HOSTINSTALLDIR` on line 44.

**Comment by ryanking13** [comment]
This issue has been fixed in #2351. Note that current stable version of `pyodide-build` released in PyPI (`0.19.1`) does not include this fix, so you'll have to install `pyodide-build` from this repository directly. I'll close this as resolved.
